src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidTestCommand.cs

```diff
@@ -120,6 +143,36 @@ protected override Task<ExitCode> InvokeInternal(ILogger logger)
             return Task.FromResult(ExitCode.GENERAL_FAILURE);
         }
 
+        private string GetDeviceToUse(ILogger logger, AdbRunner runner, string apkRequiredArchitecture)
+        {
+            var allDevicesAndTheirArchitectures = runner.GetAttachedDevicesAndArchitectures();
+            if (allDevicesAndTheirArchitectures.Count == 1)
+            {
+                // There's only one device, so -s argument isn't needed but still check
+                if (!allDevicesAndTheirArchitectures.Values.Contains(apkRequiredArchitecture, StringComparer.InvariantCultureIgnoreCase))
+                {
+                    logger.LogError($"Single device available has architecture '{allDevicesAndTheirArchitectures.Values.First()}', != '{apkRequiredArchitecture}'. Package installation will likely fail, but we'll try anyways."); 
+                }
+                return ""; // Empty = Use default device
+            }
+            else if (allDevicesAndTheirArchitectures.Count > 1)
+            {
+                if (allDevicesAndTheirArchitectures.Any(kvp => kvp.Value.Equals(apkRequiredArchitecture, StringComparison.OrdinalIgnoreCase)))
+                { 
+                    var firstAvailableCompatible = allDevicesAndTheirArchitectures.Where(kvp => kvp.Value.Equals(apkRequiredArchitecture, StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
+                    logger.LogDebug($"Using first-found compatible device - serial: '{firstAvailableCompatible.Key}' - Arch: {firstAvailableCompatible.Value}");
+                    return firstAvailableCompatible.Key;
+                }
+                else
+                {
+                    logger.LogError($"No devices found with architecture '{apkRequiredArchitecture}'.  Just returning first available device; installation will likely fail, but we'll try anyways.");
+                    return allDevicesAndTheirArchitectures.Keys.First();
+                }
+            }
+            logger.LogError("Unable to detect an attached device.");
+            return "";
```

Comment:
Same as above - `null` would be better

```suggestion
            return string.Empty;
```